File: app/src/mock/java/workshop1024/com/xproject/model/Injection.java
package workshop1024.com.xproject.model;

import workshop1024.com.xproject.model.filter.source.FilterDataSource;
import workshop1024.com.xproject.model.filter.source.FilterMockRepository;
import workshop1024.com.xproject.model.message.source.MessageDataSource;
import workshop1024.com.xproject.model.message.source.MessageMockRepository;
import workshop1024.com.xproject.model.news.source.NewsDataSource;
import workshop1024.com.xproject.model.news.source.NewsMockRepository;
import workshop1024.com.xproject.model.publisher.source.PublisherDataSource;
import workshop1024.com.xproject.model.publisher.source.PublisherMockRepository;
import workshop1024.com.xproject.model.publishertype.source.PublisherTypeDataSource;
import workshop1024.com.xproject.model.publishertype.source.PublisherTypeMockRepository;
import workshop1024.com.xproject.model.subinfo.source.SubInfoDataSource;
import workshop1024.com.xproject.model.subinfo.source.SubInfoMockRepository;

public class Injection {
    public static PublisherTypeDataSource provideContentTypeRepository() {
        return PublisherTypeMockRepository.getInstance();
    }

    public static FilterDataSource provideFilterRepository() {
        return FilterMockRepository.getInstance();
    }

    public static MessageDataSource provideMessageRepository() {
        return MessageMockRepository.getInstance();
    }

    public static NewsDataSource provideNewsRepository() {
        return NewsMockRepository.getInstance();
    }

    public static PublisherDataSource providePublisherRepository() {
        return PublisherMockRepository.getInstance();
    }

    public static SubInfoDataSource provideSubInfoRepository() {
        return SubInfoMockRepository.getInstance();
    }
}

File: main/src/mock/java/workshop1024/com/xproject/main/model/Injection.kt
package workshop1024.com.xproject.main.model

import android.content.Context
import workshop1024.com.xproject.base.utils.ExecutorUtils
import workshop1024.com.xproject.main.model.filter.source.FilterDataSource
import workshop1024.com.xproject.main.model.filter.source.FilterRepository
import workshop1024.com.xproject.main.model.filter.source.local.FilterDatabase
import workshop1024.com.xproject.main.model.filter.source.local.FilterLocalDataSource
import workshop1024.com.xproject.main.model.filter.source.remote.FilterRemoteDataSource
import workshop1024.com.xproject.main.model.publisher.local.PublisherDatabase
import workshop1024.com.xproject.main.model.publisher.local.PublisherLocalDataSource
import workshop1024.com.xproject.main.model.publisher.remote.PublisherRemoteDataSource
import workshop1024.com.xproject.main.model.publisher.source.PublisherDataSource
import workshop1024.com.xproject.main.model.publisher.source.PublisherRepository

object Injection {

    fun providePublisherRepository(context: Context): PublisherDataSource {
        return PublisherRepository.getInstance(PublisherRemoteDataSource.instance,
                PublisherLocalDataSource.getInstance(PublisherDatabase.getInstance(context).publisherDao(), ExecutorUtils()))
    }

    fun provideFilterRepository(context: Context): FilterDataSource {
        return FilterRepository.getInstance(FilterRemoteDataSource.instance,
                FilterLocalDataSource.getInstance(FilterDatabase.getInstance(context).filterDao(), ExecutorUtils()))
    }
}